src/ui/public/filter_bar/filter_bar.js

```diff
@@ -57,24 +58,14 @@ define(function (require) {
           }
         };
 
-        var privateFieldRegexp = /(^\$|meta)/;
         $scope.startEditingFilter = function (source) {
-          var model = _.cloneDeep(source);
-          var filterType;
-
-          //Hide private properties and figure out what type of filter this is
-          _.each(model, function (value, key) {
-            if (key.match(privateFieldRegexp)) {
-              delete model[key];
-            } else {
-              filterType = key;
-            }
-          });
-
-          $scope.editingFilter = {
+          return $scope.editingFilter = {
             source: source,
-            type: filterType,
-            model: model
+            type: _.first(_.filter(_.keys(source), function (key) {
```

Comment:
If I'm reading this correctly, your intention here is to return the first key that does not match the regex, and I think `_.findKey()` would be more appropriate from a performance and expressiveness standpoint:

```js
type: _.findKey(source, (val, key) => !key.match(privateFilterFieldRegex)),
```